# HG changeset patch
# User David Meyer <pdox@google.com>
# Date 1305932917 25200
# Branch pnacl-sfi
# Node ID 9747397423656946cbb9324cef3fe9d5c74a5a58
# Parent b5512fb0c3a9685217eb6f604280cc9e55a2d2f9
Make va_list 16-bytes instead of 24-bytes in llvm-gcc

 From llvm-gcc-0005-78-9747397423656946cbb9324cef3fe9d5c74a5a58.patch

diff -r b5512fb0c3a9 gcc/tree.c
--- a/gcc/tree.c	Wed May 18 13:53:00 2011 -0700
+++ b/gcc/tree.c	Wed Aug 10 10:52:06 2011 -0700
@@ -6749,11 +6749,11 @@
       {
         /* If not expanding va_arg, make va_list large enough
            so that the generated bitcode will work on all platforms.
-           (e.g., X86-64 needs exactly 24-bytes for va_list) */
+           (in particular, NaCl X86-64 needs exactly 16-bytes for va_list) */
 
         t = build_array_type (
-                long_long_unsigned_type_node,
-                build_index_type (build_int_cst (NULL_TREE, 3-1)));
+                unsigned_type_node,
+                build_index_type (build_int_cst (NULL_TREE, 4-1)));
       }
     else
         t = targetm.build_builtin_va_list ();
